### PR TITLE
chore: upgrade React from 16 to 18

### PR DESCRIPTION
**Pull request by naftis**
This PR upgrades React version to 18

I have followed React 16 -> 17 upgrade guide (no actions needed to be taken) and 17 -> 18 upgrade guide (children -property now needs to be explicit & React app is initialised with `createRoot`)

### CODE REVIEW

**Comment by naftis** [comment]
@Zangetsu101 there's only 2 failing tests here. Wasn't able to quickly figure out what's the deal with them.
```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 2 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/views/SysAdmin/Team/TeamSearch.test.tsx > Team search test > Team search with location in props > loads the location in the search input box
Error: Method “props” is meant to be run on 1 node. 0 found instead.
 ❯ ReactWrapper.single ../../node_modules/enzyme/src/ReactWrapper.js:1168:13
 ❯ ReactWrapper.props ../../node_modules/enzyme/src/ReactWrapper.js:710:17
 ❯ src/views/SysAdmin/Team/TeamSearch.test.tsx:110:54
    108|     it('loads the location in the search input box', () => {
    109|       expect(
    110|         app.find('#locationSearchInput').hostNodes().props().value
       |                                                      ^
    111|       ).toEqual('Alokbali Union Parishad')
    112|     })
 ❯ runNextTicks internal/process/task_queues.js:60:5
 ❯ processTimers internal/timers.js:497:9



⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/2]⎯

 FAIL  src/views/SysAdmin/Team/user/UserAuditActionModal.test.tsx > user audit action modal tests > in case of successful deactivate audit action > after filling mandatory data > clicking confirm action dispatches success notification action
AssertionError: expected false to be true // Object.is equality
 ❯ src/views/SysAdmin/Team/user/UserAuditActionModal.test.tsx:262:11
    260|         expect(
    261|           store.getState().notification.userAuditSuccessToast.visible
    262|         ).toBe(true)
       |           ^
    263|       })
    264|     })

  - Expected   "true"
  + Received   "false"
  ```

**Comment by naftis** [comment]
Thank you Tameem for fixing the unit tests. TypeScript compilation with `tsc` is still failing though as I forgot to check it earlier. Lets fix that when we find a good time for that

File: packages/client/src/views/OfficeHome/inProgress/inProgress.test.tsx [hunk 1]
```diff
@@ -902,15 +904,17 @@ describe('In Progress tab', () => {
...
         )
 
-        expect(data[0].id).toBe('f0a1ca2c-6a14-4b9e-a627-c3e2e110587e')
-        expect(data[0].name).toBe('anik hoque')
-        expect(data[0].notificationSent).toBe(EXPECTED_DATE_OF_REJECTION)
-        expect(data[0].event).toBe('Birth')
+        expect(data[1].id).toBe('f0a1ca2c-6a14-4b9e-a627-c3e2e110587e')
+        expect(data[1].name).toBe('anik hoque')
+        expect(data[1].notificationSent).toBe(EXPECTED_DATE_OF_REJECTION)
+        expect(data[1].event).toBe('Birth')
```
Comment (naftis) [inline packages/client/src/views/OfficeHome/inProgress/inProgress.test.tsx:917]: @Zangetsu101 wonder why the Birth is now `[1]` of the array and not `[0]` like before. Like the rendering order would have changed for some reason.... 🤔

File: packages/client/src/views/OfficeHome/readyForReview/readyForReview.test.tsx [hunk 2]
```diff
@@ -303,19 +303,19 @@ describe('OfficeHome sent for review tab related tests', () => {
...
-    expect(data[0].trackingId).toBe('BW0UTHR')
-    expect(data[0].event).toBe('Birth')
-    expect(data[0].actions).toBeDefined()
+    expect(data[1].id).toBe('9a55d213-ad9f-4dcd-9418-340f3a7f6269')
+    expect(data[1].dateOfEvent).toBe('8 years ago')
+    expect(data[1].sentForReview).toBe(EXPECTED_DATE_OF_DECLARATION)
+    expect(data[1].name).toBe('iliyas khan')
+    expect(data[1].trackingId).toBe('BW0UTHR')
+    expect(data[1].event).toBe('Birth')
+    expect(data[1].actions).toBeDefined()
```
Comment (naftis) [inline packages/client/src/views/OfficeHome/readyForReview/readyForReview.test.tsx:318]: Same here. I'm not sure why the index changed

File: packages/client/src/views/OfficeHome/inProgress/inProgress.test.tsx (hunk 1 above)
Comment (naftis) [reply packages/client/src/views/OfficeHome/inProgress/inProgress.test.tsx]: Fixed in commit [97efb36](https://github.com/opencrvs/opencrvs-core/pull/4587/commits/97efb366da85a428526d032e7b38d23b47590cca)

File: packages/client/src/views/RegisterForm/review/ReviewSection.tsx [hunk 3]
```diff
@@ -1171,51 +1171,53 @@ class ReviewSectionComp extends React.Component<FullProps, State> {
...
-                ? this.getFieldValueWithErrorMessage(
-                    section,
-                    field,
-                    errorsOnNestedField[0]
-                  )
-                : nestedValue}
-            </>
-          )
-        }, <>{value}</>)
-      : value
```
Comment (naftis) [inline packages/client/src/views/RegisterForm/review/ReviewSection.tsx:1218]: The essential change here, is that I wrapped the `: value` to be `<>{value}</>` to always return a ReactNode from the method.

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**